iconic/UserProfile/views.py

```diff
@@ -173,17 +174,76 @@ def put(self, request):
         else:
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
- # Method for deliting city by id  
+    # Method for deliting city by id
     def delete(self, request):
         Cities.objects.get(id=request.data['id']).delete()
         return Response(status=status.HTTP_200_OK)
 
 
-#View for getting all cities
+# View for getting all cities
 class CitiesGet(APIView):
-    
-#Method for getting all cities
+
+    # Method for getting all cities
     def get(self, request):
         cities = Cities.objects.all()
         serializer = CitiesSerializer(cities, many=True)
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+# View for getting comment! adding, getting, changing comment by authorize user
+class CommentEdit(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    # Method for getting comment by id
+    def get(self, request):
+        comment = Comment.objects.get(id=request.data['id'])
+        serializer = CommentSerializer(comment)
+        if serializer.is_valid():
+            return Response(serializer.data,
+                            status=status.HTTP_200_OK)
+        else:
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
```

Comment:
I now thought, and decided that this request must be unauthorized
Make:
comments/ url for this one
comments/all/ for getting all comments of the post
comments/edit/ for post,put and delete
Also you do not need to check if the data is valid (line 201) just return Response(serializer.data, status=status.HTTP_200_OK)